### DISCUSSION TITLE
Upsert using array

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#9701] [General]
Using below to upsert in SupaBase/JS:

```
const{ data,error }=await supabase
.from('DB')
.upsert(
{"A":"NOW()","B":"hhh"} , { onConflict:'B,C,D' } ,
)
if(error){ console.log(error); }
await supabase
.from('DB')
.select()
if(error){ console.log(error); }

```

How can you do above using an array of rows, as to limit the requests to one for upserting multiple rows.

Eg:

```
let thearray='{"A":"NOW()","B":"hhh"},{"A":"NOW()","B":"hhh"},{"A":"NOW()","B":"hhh"},{"A":"NOW()","B":"hhh"}';
const{ data,error }=await supabase
.from('DB')
.upsert(
thearray , { onConflict:'B,C,D' } ,
)
if(error){ console.log(error); }
await supabase
.from('DB')
.select()
if(error){ console.log(error); }

```

I keep getting "All object keys must match"

**Comment by GaryAustin1** [comment]
Looks like thearray is a string.   At a minimum you need to do let thearray = [{},{},{}].
You may also hit another issue, but I can't tell from your descriptions what you are trying to do.
Might also be related, but I don't think you are there yet.  https://github.com/supabase/postgrest-js/issues/173

**Reply by ghost** [reply]
Thank you. Indeed in my case it was simply a matter of poorly formatted array.
